mpf/devices/driver.py

```diff
@@ -318,3 +347,33 @@ def pulse(self, pulse_ms: int = None, pulse_power: float = None, max_wait_ms: in
             self._pulse_now(pulse_ms, pulse_power)
 
         return wait_ms
+
+    @event_handler(4)
+    def event_timed_enable(self, hold_ms: int = None, hold_power: float = None, pulse_ms: int = None, pulse_power: float = None,  max_wait_ms: int = None, **kwargs):
+        """Event handler for timed enable events."""
+        del kwargs
+        self.timed_enable(hold_ms, hold_power, pulse_ms, pulse_power, max_wait_ms)
+
+    def timed_enable(self, hold_ms: int = None, hold_power: float = None, pulse_ms: int = None, pulse_power: float = None, max_wait_ms: int = None):
+        """Pulse and enable this driver.
+
+        Args:
+        ----
+            enable_duration_ms: The number of milliseconds the driver should be
+                enabled for. If no value is provided, the driver will be enabled
+                for the value specified in the config dictionary (default_hold_ms).
+            pulse_ms: The number of seconds the driver should be enabled at pulse_power
+            pulse_power: The pulse power for the initial pulse. A float between 0.0 and 1.0
+            hold_power: The hold power for the enable duration. A float between 0.0. and 1.0
+            max_wait_ms: Maximum time this pulse may be delayed for PSU optimization
+        """
+        pulse_duration = self.get_and_verify_pulse_ms(pulse_ms)
+        pulse_power = self.get_and_verify_pulse_power(pulse_power)
+        hold_duration = self.get_and_verify_hold_ms(hold_ms)
+        hold_power = self.get_and_verify_hold_power(hold_power)
+        wait_ms = self._notify_psu_and_get_wait_ms(pulse_duration, max_wait_ms)
+
+        self.hw_driver.timed_enable(PulseSettings(pulse_power, pulse_duration),
```

Comment:
We probably want a try catch here for two cases:
a) Hardware does not support timed_enable
b) Hardware does not support the duration we need (i.e. only up to 500ms for Spike)

In both cases we need to emulate in software.

Reply:
Should we have a log entry to inform the user that their platform doesn't support timed_enable and we are falling back to software?

Reply:
Yes but we need to be careful not to spam too much as this might happen super often. Maybe at startup or somewhere?